src/coreclr/jit/gentree.cpp

```diff
@@ -19573,9 +19681,22 @@ GenTree* Compiler::gtNewSimdBinOpNode(genTreeOps  op,
         case GT_RSH:
         case GT_RSZ:
         {
-            assert(!varTypeIsFloating(simdBaseType));
```

Comment:
I see that `Vector*<T>` is going to support shifting so it makes sense that the restriction is lifted, then you normalize the base type.